### PR TITLE
Woche6

### PR DESCRIPTION
**Pull request by Koch-Dustin**
Changes:
   Add dna
   Add hello
   Add mario-less
   Add mario-more
   Add cash
   Add credit
   Add readability

PR #8

### CODE REVIEW

File: Woche6/dna/dna.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    with open(db_path, "r") as csvfile:
+        file_reader = DictReader(csvfile)
+        dict_list = list(file_reader)
+
+    with open(seq_path, "r") as file:
+        sequence = file.read()
+
+    sequence
+    max_counts = []
+    for i in range(1, len(reader.fieldnames)):
```
Comment (MKrips) [inline Woche6/dna/dna.py:23]:
```suggestion
    for i in range(1, len(file_reader.fieldnames)):
```
This should be the specific CsvReader that read the file, not any generic one.

File: Woche6/dna/dna.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            str_count = 0
+
+            if sequence[j:(j + len(str))] == str:
+                second_char_to_check_if_repeat = 0
+                while sequence[(j + second_char_to_check_if_repeat):(j + second_char_to_check_if_repeat + len(str))] == str:
+                    str_count += 1
+                    second_char_to_check_if_repeat += len(str)
+
+                if str_count > max_counts[i - 1]:
+                    max_counts[i - 1] = str_count
```
Comment (MKrips) [inline Woche6/dna/dna.py:37]: This is what you are supposed to do the longest_match function for.

File: Woche6/dna/dna.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            if matches == len(reader.fieldnames) - 1:
+                print(dict_list[i]['name'])
+                exit(0)
+
+    print("No match")
+
+    return
+
+
+def longest_match(sequence, subsequence):
```
Comment (MKrips) [inline Woche6/dna/dna.py:56]: This function is not called, making it currently unnecessary.

File: Woche6/dna/dna.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    longest_run = 0
+    subsequence_length = len(subsequence)
+    sequence_length = len(sequence)
+
+    for i in range(sequence_length):
+
+        count = 0
+
+        while True:
```
Comment (FrederikFricke) [inline Woche6/dna/dna.py:66]:
```suggestion
    for i in range(sequence_length):
        count = 0

        while True:
```

File: Woche6/mario-less/main.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+from cs50 import get_int
+
+
+def input_number(message):
```
Comment (MKrips) [inline Woche6/mario-less/main.py:4]: You never use this function

File: Woche6/dna/dna.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+            start = i + count * subsequence_length
+            end = start + subsequence_length
+
+            if sequence[start:end] == subsequence:
+                count += 1
+
+            else:
+                break
```
Comment (FrederikFricke) [inline Woche6/dna/dna.py:75]:
```suggestion
            if sequence[start:end] == subsequence:
                count += 1
            else:
                break
```

File: Woche6/mario-less/main.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+
+height = get_int("Height: ")
+while height < 1 or height > 8:
+    height = get_int("Height: ")
+
+for i in range(height):
+    print(" " * (height-1-i), end="")
+    print("#" * (i+1), end="  ")
+    print(" " * (height-1-i))
```
Comment (MKrips) [inline Woche6/mario-less/main.py:21]:
```suggestion
    print("#" * (i+1), end="  ")
```

**Review by FrederikFricke — CHANGES_REQUESTED** [review]

File: Woche6/mario-more/mario-more.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+height = input_number("Height: ")
+while height < 1 or height > 8:
+    height = get_int("Height: ")
+
+for i in range(height):
+    print(" " * (height-1-i), end="")
+    print("#" * (i+1), end="  ")
+    print("#" * (i+1), end="")
+    print(" " * (height-1-i))
```
Comment (MKrips) [inline Woche6/mario-more/mario-more.py:22]:
```suggestion
    print("#" * (i+1), end="")
```

File: Woche6/sentimental-credit/credit.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+workingCC = credit_card_number
+
+while workingCC > 0:
+    last_digit = workingCC % 10
+    separate_digits = separate_digits + last_digit
+    workingCC = workingCC // 100
+
+
+workingCC = credit_card_number // 10
```
Comment (MKrips) [inline Woche6/sentimental-credit/credit.py:34]:
```suggestion
workingCC = credit_card_number // divisor
```
If you defined a variable, do not just use it sometimes, but everytime you can

File: Woche6/sentimental-credit/credit.py [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+while i < count - 2:
+    divisor = divisor * 10
+    i += 1
+
+
+first_digit = credit_card_number // divisor
+first_two_digits = credit_card_number // (divisor // 10)
+
+
+def check_which_type_card_is():
```
Comment (MKrips) [inline Woche6/sentimental-credit/credit.py:57]: Do not declare functions randomly in the middle, declare them before your code

**Review by MKrips — CHANGES_REQUESTED** [review]

File: Woche6/mario-more/mario-more.py [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+
+height = input_number("Height: ")
+while height < 1 or height > 8:
+    height = get_int("Height: ")
+
+for i in range(height):
+    print(" " * (height-1-i), end="")
+    print("#" * (i+1), end="  ")
+    print("#" * (i+1), end="")
```
Comment (stefanluth) [inline Woche6/mario-more/mario-more.py:21]:
```suggestion
    print(" " * (height - 1 - i), end="")
    print("#" * (i + 1), end="  ")
    print("#" * (i + 1), end="")
```

File: Woche6/sentimental-credit/credit.py [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+        except ValueError:
+            continue
+        else:
+            return user_input
+
+
+def check_which_type_card_is():
+    if separate_digits % 10 == 0:
+        card_is_master_card = count == 16 and (first_two_digits == 51 or first_two_digits == 52 or first_two_digits == 53 or first_two_digits == 54 or first_two_digits == 55)
```
Comment (stefanluth) [inline Woche6/sentimental-credit/credit.py:21]:
```suggestion
        card_is_master_card = count == 16 and (
            first_two_digits == 51
            or first_two_digits == 52
            or first_two_digits == 53
            or first_two_digits == 54
            or first_two_digits == 55
        )
```

File: Woche6/sentimental-readability/readability.py [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+s = (sentences * (100 / words))
+
+index = round(((0.0588 * L) - (0.296 * s) - 15.8))
+
+if index < 1:
+    print("Before Grade 1")
+elif index > 16:
+    print("Grade 16+")
+else:
+    print(f'Grade {index}')
```
Comment (stefanluth) [inline Woche6/sentimental-readability/readability.py:30]:
```suggestion
    print(f"Grade {index}")
```

**Review by stefanluth — CHANGES_REQUESTED** [review]

**Review by DevBastiS — APPROVED** [review]

**Review by stefanluth — APPROVED** [review]

**Pull request merged**